src/tritonserver.cc

```diff
@@ -2755,6 +2755,38 @@ TRITONSERVER_ServerModelStatistics(
     for (const auto& version : mv_pair.second) {
       std::shared_ptr<tc::Model> model;
       RETURN_IF_STATUS_ERROR(lserver->GetModel(mv_pair.first, version, &model));
+
+      // Add memory usage
+      triton::common::TritonJson::Value memory_usage(
```

Comment:
Question: In case memory tracking isn't enabled - we include an empty array is that correct? Is it possible to tell the difference between tracking enabled and value being 0 (i.e. no memory reported?)

Reply:
Good question, the backend can report 0 value to be explicit that no memory will be released for the corresponding memory type. It is probably safer to say that the memory type not being included / reported only means it is not tracked, and it may be released somehow ("pessimistic" estimation). In other words, the value reported is what the backend is confident to release on model / instance unloading.